src/rubrix/client/datasets.py

```diff
@@ -17,7 +17,9 @@
 from typing import Any, Dict, List, Optional, Tuple, Type, Union
 
 import pandas as pd
+import spacy
```

Comment:
The optional spacy modules must be imported locally, handling errors when the modules are not found.

Reply:
Apparently, if I take out import spacy, even if from spacy.tokens import DocBin is still there, I got an error with the type hinting of the function:

```
(rubrix) ➜  rubrix git:(spacy_docbin) ✗ /Users/ignacio/opt/anaconda3/envs/rubrix/bin/python /Users/ignacio/Documents/Recognai/rubrix/t
esting_docbin.py
Traceback (most recent call last):
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/utils.py", line 111, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/Users/ignacio/opt/anaconda3/envs/rubrix/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/client/__init__.py", line 16, in <module>
    from .rubrix_client import RubrixClient
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/client/rubrix_client.py", line 29, in <module>
    from rubrix.client.datasets import (
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/client/datasets.py", line 645, in <module>
    class DatasetForTokenClassification(DatasetBase):
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/client/datasets.py", line 769, in DatasetForTokenClassification
    ) -> Union["datasets.Dataset", spacy.tokens.DocBin]:
NameError: name 'spacy' is not defined

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/ignacio/Documents/Recognai/rubrix/testing_docbin.py", line 4, in <module>
    dataset = rb.DatasetForTokenClassification([
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/utils.py", line 81, in __getattr__
    module = self._get_module(self._class_to_module[name])
  File "/Users/ignacio/Documents/Recognai/rubrix/src/rubrix/utils.py", line 113, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import rubrix.client.datasets because of the following error (look up to see its traceback):
name 'spacy' is not defined
```

Reply:
You should replace the line:

```python
Union["datasets.Dataset", spacy.tokens.DocBin]
```
by 
```python
Union["datasets.Dataset", "spacy.tokens.DocBin"]
```
Including the `"` should fix your problem.